applications/vessel-history/src/features/search/search.thunk.ts

```diff
@@ -1,19 +1,71 @@
 import { createAsyncThunk } from '@reduxjs/toolkit'
-import GFWAPI from '@globalfishingwatch/api-client'
+import { stringify } from 'qs'
+import GFWAPI, {
+  AdvancedSearchQueryField,
+  getAdvancedSearchQuery,
+} from '@globalfishingwatch/api-client'
 import { VesselSearch } from '@globalfishingwatch/api-types'
 import { BASE_DATASET, RESULTS_PER_PAGE, SEARCH_MIN_CHARACTERS } from 'data/constants'
 import { RootState } from 'store'
 import { CachedVesselSearch } from './search.slice'
 
-const fetchData = async (query: string, offset: number, signal?: AbortSignal | null) => {
-  return await GFWAPI.fetch<any>(
-    `/v1/vessels/search?datasets=${encodeURIComponent(
-      BASE_DATASET
-    )}&limit=${RESULTS_PER_PAGE}&offset=${offset}&query=${encodeURIComponent(query)}&useTMT=true`,
+export const getSerializedQuery = (query: string, advancedSearch?: Record<string, any>) => {
+  if (!advancedSearch) return query
+  const fields: AdvancedSearchQueryField[] = [
     {
-      signal,
-    }
-  )
+      key: 'shipname',
+      value: query,
+    },
+    {
+      key: 'mmsi',
+      value: advancedSearch.mmsi,
+    },
+    {
+      key: 'imo',
+      value: advancedSearch.imo,
+    },
+    {
+      key: 'callsign',
+      value: advancedSearch.callsign,
+    },
+    {
+      key: 'flag',
+      value: advancedSearch.flags.map((f: string) => ({ id: f })),
+    },
+    {
+      key: 'lastTransmissionDate',
+      value: advancedSearch.lastTransmissionDate,
+    },
+    {
+      key: 'firstTransmissionDate',
+      value: advancedSearch.firstTransmissionDate,
+    },
```

Comment:
Review with @rrequero the fields that we should send to the advanced search endpoint when searching TMT > https://globalfishingwatch.atlassian.net/browse/VV-187?focusedCommentId=13604